test/dynamo/test_subclasses.py

```diff
@@ -1696,6 +1871,664 @@ def fn(t0, t1, t2):
+        return (view_1,)
+""",  # noqa: B950
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(bw[1].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_3: "Sym(s0)", tangents_1: "f32[s0]"):
+        view_1: "f32[s0]" = torch.ops.aten.view.default(tangents_1, [primals_3]);  tangents_1 = None
+        return (view_1, None, primals_3)
+""",  # noqa: B950
+        )
+
+    def test_tensor_subclass_WrapperTensor_mark_dynamic_shapes(self):
+        def f(t):
+            y = t.clone()
+            return y.view(a.shape[1] * a.shape[0]), y.a
+
+        a = torch.ones(3, 4, requires_grad=True)
+        t = WrapperTensor(a)
+        torch._dynamo.mark_dynamic(t, 0)
+
+        fw, bw = self._compile_check(
+            f,
+            [
+                (t,),
+            ],
+            dynamic=None,
+            call_backward=True,
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(fw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_1: "f32[s0, 4]", primals_2: "Sym(s0)", primals_3: "Sym(s0)", primals_4: "Sym(s0)"):
+        clone: "f32[s0, 4]" = torch.ops.aten.clone.default(primals_1);  primals_1 = None
+
+        mul: "Sym(4*s0)" = 4 * primals_2;  primals_2 = None
+        view: "f32[4*s0]" = torch.ops.aten.view.default(clone, [mul])
+        return (view, clone, mul, primals_4)
+""",  # noqa: B950
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(bw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_4: "Sym(s0)", tangents_1: "f32[4*s0]"):
+        view_1: "f32[s0, 4]" = torch.ops.aten.view.default(tangents_1, [primals_4, 4]);  tangents_1 = None
+        return (view_1, None, None, primals_4)
+""",  # noqa: B950
+        )
+
+    @unittest.expectedFailure
+    def test_tensor_subclass_DoubleTensor_simple(self):
+        def f(t):
+            y = t.clone()
+            return y.view(-1) * 2
+
+        a = torch.ones(3, 4, requires_grad=True)
+        t = DoubleSizeTensor(a)
+
+        fw, bw = self._compile_check(f, [(t,)], dynamic=True, call_backward=True)
+
+        self.assertExpectedInline(
+            normalize_gm(fw[0].print_readable(print_output=False)),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_1: "f32[s2, s1]", primals_2: "Sym(s2)", primals_3: "Sym(s1)", primals_4: "Sym(2*s2)", primals_5: "Sym(s1)"):
+        mul: "f32[s2, s1]" = torch.ops.aten.mul.Tensor(primals_1, 2);  primals_1 = None
+        return [mul, primals_4, primals_5, primals_3, primals_2]
+""",  # noqa: B950
+        )
+
+        self.assertExpectedInline(
+            normalize_gm(bw[0].print_readable(print_output=False)),
+            """\
+    # code below is wrong!
```

Comment:
where is this comment coming from 🤔

Reply:
Oh, I wrote this! The test is annotated with `unittest.expectedFailure`.